src/routes/resources/resources.handlers.ts

```
@@ -1082,3 +1201,391 @@ export const upvoteJobBatch = async (c: Context) => {
+    const resourceIds = userBookmarksWithResources.map((r) => r.resource.id);
+
+    // Batch Redis operations in a single pipeline
+    const pipeline = redis.pipeline();
+
+    // 1. Get all upvote counts
+    const countKeys = resourceIds.map(getUpvoteCountKey);
+    pipeline.mget(...countKeys);
+
+    // 2. Get user upvote status in the same pipeline if user is logged in
+    const upvoteKeys = resourceIds.map(
+      (id) => `upvote:user:${userLoggedIn.id}:resource:${id}`
```

ESLint

[error] Unexpected parentheses around single function argument having a body with no curly braces.

(style/arrow-parens)